components/inputs/input-date.js

```diff
@@ -404,10 +404,15 @@ class InputDate extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(LitEl
 		if (this._dropdown && !this._dropdown.openedAbove && !fullCalendarVisible) {
 			this._dropdown.querySelector('d2l-calendar').scrollIntoView({ block: 'nearest', behavior: 'smooth', inline: 'nearest' });
 		}
-		// use setTimeout to wait for keyboard to open on mobile devices
-		setTimeout(() => {
-			this._textInput.scrollIntoView({ block: 'nearest', behavior: 'smooth', inline: 'nearest' });
-		}, 150);
+
+		// if keyboard opens (in all cases except mobile calendar view),
+		// ensure text input is in viewport
+		if (!(mediaQueryList.matches && this._dropdown.opened)) {
```

Comment:
Just to clarify since I've been out of this code for a bit - would this if statement ever be true? I think `_handleDropdownOpen` is only ever called after `opened` is set to true.

Reply:
I made a recent change here that prevents opening the keyboard on mobile if the calendar is open. It's a bit confusing here so I've added that comment on top so we can remember the reasoning. So if we are in mobile AND the calendar is open, the keyboard will never open, so do not scroll the input into view. All other cases (non-mobile, mobile with keyboard), this will be true.

Reply:
Ah, you are right in the mobile case that the keyboard and non-calendar opening flow would never get here... because the dropdown would never open 🤦. This code should still work, but it has the redundant this._dropdown.opened

Reply:
Ah okay yup that was what I was wondering about - thanks for clarifying! Otherwise LGTM